apps/wallet-mobile/src/features/Swap/common/SwapFormProvider.tsx

```diff
@@ -1,25 +1,58 @@
-import React, {createContext, ReactNode, useContext, useReducer} from 'react'
+import {useSwap} from '@yoroi/swap'
+import React from 'react'
+import {Alert, TextInput} from 'react-native'
 
-type TouchedState = {
-  isSellTouched: boolean
-  isBuyTouched: boolean
-  isPoolTouched: boolean
+import {useLanguage} from '../../../i18n'
+import {useSelectedWallet} from '../../../SelectedWallet'
+import {useBalances, useTokenInfo} from '../../../yoroi-wallets/hooks'
+import {Amounts, Quantities} from '../../../yoroi-wallets/utils'
+import {useStrings} from './strings'
+
+type SwapFormState = {
+  sellAmount: {
+    isTouched: boolean
+    disabled: boolean
+    error: string | undefined
+    displayValue: string
+  }
+  buyAmount: {
+    isTouched: boolean
+    disabled: boolean
+    error: string | undefined
+    displayValue: string
+  }
+  selectedPool: {
+    isTouched: boolean
+  }
 }
-type TouchedActions = {
+type SwapFormActions = {
   sellTouched: () => void
   buyTouched: () => void
   switchTouched: () => void
   poolTouched: () => void
   poolDefaulted: () => void
   resetTouches: () => void
+  setBuyInputValue: (value: string) => void
+  setSellInputValue: (value: string) => void
+  setBuyAmountError: (error: string | undefined) => void
+  setSellAmountError: (error: string | undefined) => void
```

Comment:
for consistency with other reducers
```suggestion
  buyInputValueChanged: (value: string) => void
  sellInputValueChanged: (value: string) => void
  buyAmountErrorChanged: (error: string | undefined) => void
  sellAmountErrorChanged: (error: string | undefined) => void
```

Reply:
reseTouches will become resetSwapForm